src/sql/base/browser/ui/panel/panel.ts

```diff
@@ -173,6 +177,26 @@ export class TabbedPanel extends Disposable implements IThemable {
 		}
 		this._tabMap.get(tab).header.destroy();
 		this._tabMap.delete(tab);
+		if (this._shownTab === tab) {
+			this._shownTab = undefined;
+			while (this._shownTab === undefined && this.tabHistory.length > 0) {
+				let lastTab = this.tabHistory.shift();
```

Comment:
Is it possible for the tab history to contain a tab that has already been removed? If so, is that case handled correctly?

Reply:
Thats what the next line is checking. When a tab is removed it is removed from the `this._tabMap`, which means the if case will fail, and the while loop will continue.